src/metro/devices/display/fast_plot.py

```diff
@@ -1268,6 +1271,9 @@ def dataAdded(self, d):
         if self.idx_data.shape[1] == 0:
             return
 
+        self.idx_data = np.nan_to_num(self.idx_data, copy=True,
+                                      nan=nan_replacement)
+
```

Comment:
Note: would like to avoid copy.
Cannot replace in-place as destination is not writable.
Ideally, would only copy if any value needs replacing - but first checking that and then copying doesn't seem clever either.

Also: I considered if we could drop instead of replacing.
However, if `self.idx_data.shape[0] > 1`, then we cannot do a simple 1d mask and `np.compress` on x and ys.
And stacking seems tricky.
Unless we drop by logical OR across `axis=0`.

Reply:
Hmmm, this is likely due to the no-copy mechanism used with `pyzmq`. Generally though, I'd also like to avoid encoding `metropc`-specific details in the display devices.

We could either implement it directly in the rendering code, replacing `NaN` by something else, or move this specific into the `euxfel.metropc` device. Another advantage of the rendering code would be that we can actually render gaps rather than replace it. And one could present `np.inf` by an infinite slope before and after.